### ISSUE TITLE
Not able to achieve same results as the paper for RAF-DB

### ISSUE THREAD

**Issue by ipa-anm-sy** [simon20010923/DDAMFN#9]
I am getting the best accuracy upto 89.899 with rafdb dataset, How can I achieve the results from the paper.

**Comment by simon20010923** [comment]
You can use this Sharpness aware minimization for efficiently improving generalization, an optimizer for training.

**Comment by ipa-anm-sy** [comment]
While using predictions on a live camera feed, with gpu, my fps is only 4 do you have any suggestions for improving this?

**Comment by 1chenchen22** [comment]
> 我使用 rafdb 数据集获得了高达 89.899 的最佳准确度，我怎样才能从论文中获得结果。



> 您可以使用此锐度感知最小化来有效地改进泛化，这是一种用于训练的优化器。

我也是没有能达到那么高的准确率，最高也就只有百分之90，还有其他改进优化的准确率的技巧吗

**Comment by simon20010923** [comment]
I think maybe you can crop the img to a small scale, since our pretrained model pay attention on face areas and maybe your img have large space of background?

**Comment by 1chenchen22** [comment]
> I think maybe you can crop the img to a small scale, since our pretrained model pay attention on face areas and maybe your img have large space of background?

好哒谢谢你，我没有对原来的数据集做裁剪，可能就导致他的背景比较大吧,

**Comment by ipa-anm-sy** [comment]
Do you have some ideas for improvinf performance, and fps rate.

**Comment by simon20010923** [comment]
What do you mean is only 4?

**Comment by ipa-anm-sy** [comment]
I am using a live camera feed input. (640x480 pixels)
resize to 300x300 for face detection and cropping.
then resize to   
only have 4 frames per second prediction rate. And GPU utilization rate is near about 20% for prediction. So can you suggest a good image input method to the pretrained model, which will distribute the computational load over gpu memory and increase my frames per second significantly?

**Comment by ipa-anm-sy** [comment]
I am currently doing these preprocessing steps before loading the image into the model for prediction.
preprocess = transforms.Compose([
                transforms.Resize((112,112)), 
                transforms.ToTensor(),
                transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
            ])
Is it possible to load image with higher resolution for better features quality to increase the accuracy of the prediction.
If yes how much bigger image I can load in the model?
I have trained the model with same preprocessing steps. So Do I need to retrain the model with bigger transform.resize?

**Comment by simon20010923** [comment]
Maybe you can wait for our new trained model, we will post it ASAP.

**Comment by simon20010923** [comment]
Our new version DDAMFN++ is available in the folder https://github.com/simon20010923/DDAMFN/DDAMFN++, which includes new alignment code refer to https://github.com/biubug6/Pytorch_Retinaface and optimized by SAM ([Sharpness aware minimization for efficiently improving generalization](https://arxiv.org/abs/2010.01412)). The trained RAF-DB model is available in DDAMFN++/checkpoints_ver2.0, which is not our best. The affectnet model will soon be available.

**Comment by simon20010923** [comment]
Our new version DDAMFN++ is available in the folder https://github.com/simon20010923/DDAMFN/tree/main/DDAMFN%2B%2B, which includes new alignment code refer to https://github.com/biubug6/Pytorch_Retinaface and optimized by SAM ([Sharpness aware minimization for efficiently improving generalization](https://arxiv.org/abs/2010.01412)). The trained RAF-DB model is available in DDAMFN++/checkpoints_ver2.0, which is not our best. The affectnet model will soon be available.

**Comment by ipa-anm-sy** [comment]
Thank you for providing the trained model. I tested it. It is performing great compared to other models. But my emphasis is on the emotion 'Fear'. It is difficult to obtain in the real camera feed.

So for giving better features to the model can I train the model with higher resolution images? Can I change transforms.Resize((112,112)) while training the model. Have you tried this already?

**Comment by simon20010923** [comment]
Sorry, we haven't tried that before. I think maybe you have to train a pretrained model on large images.

**Comment by ipa-anm-sy** [comment]
Should I use DDAMFN/DDAMFN++/rafdb_train_sam_opt_ver2.0.py file for training DDAMFN++ model

**Comment by ipa-anm-sy** [comment]
I am getting his error while training the model
''Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. " ''
  I looked for solutions online, but I don't see any problem with the code. Can you help me with this?

**Comment by simon20010923** [comment]
I will come back to you soon.

**Comment by ipa-anm-sy** [comment]
Do you have any updates?
